vizro-core/docs/pages/user-guides/dynamic-data.md

```diff
@@ -0,0 +1,134 @@
+
+Unlike static data, dynamic data cannot be supplied directly into the `data_frame` argument of a `figure`. Instead, it must first be added to the Data Manager and then referenced by name.
+
+The below toy example demonstrates how dynamic data is updated every time the page is refreshed. When you run the code and refresh the page the data shown will change every time because the function `load_iris_data` is re-run. The example uses the Iris dataset saved to a file `iris.csv` in the same directory as `app.py`. This data can be generated using `px.data.iris()` or [downloaded](../../assets/user_guides/data/iris.csv).
+
+!!! example "Dynamic data"
+    === "app.py"
+        ```py
+        from vizro import Vizro
+        import pandas as pd
+        import vizro.plotly.express as px
+        import vizro.models as vm
+
+        from vizro.managers import data_manager
+
+        def load_iris_data():
+            iris = pd.read_csv("iris.csv") # (1)!
+            return iris.sample(30) # (2)!
+
+        data_manager["iris"] = load_iris_data # (3)!
+
+        page = vm.Page(
+            title="My first page",
+            components=[
+                vm.Graph(figure=px.scatter("iris", x="sepal_length", y="petal_width", color="species")), # (4)!
+            ],
+            controls=[vm.Filter(column="species")],
+        )
+
+        dashboard = vm.Dashboard(pages=[page])
+
+        Vizro().build(dashboard).run()
+        ```
+
+        1. `iris` is a pandas DataFrame created by reading from the CSV file `iris.csv`.
+        2. To demonstrate that dynamic data can change when the page is refreshed, select 30 points at random. This simulates what would happen if your file `iris.csv` were constantly changing.
+        3. To use `load_iris_data` as dynamic data it must be added to the Data Manager. You should **not** actually call the function as `load_iris_data()`; doing so would result in static data that cannot be reloaded.
+        4. Dynamic data is referenced by the name of the data source `"iris"`.
+
+    === "Result"
+        [![DynamicData]][DynamicData]
+
+    [DynamicData]: ../../assets/user_guides/data/dynamic_data.gif
+
+Since dynamic data sources must always be added to the Data Manager and referenced by name, they may be used in YAML configuration [exactly the same way as for static data sources](static-data.md#reference-by-name).
+
+## Configure cache
+
+By default, every time the dashboard is refreshed a dynamic data function will be executed again. This means that your dashboard will always show the very latest data. In fact, if there are multiple graphs on the same page using the same dynamic data source then the loading function will be executed _multiple_ times, once for each graph on the page. Hence, if loading your data is a slow operation, your dashboard performance may suffer.
```

Comment:
```suggestion
By default, each time the dashboard is refreshed a dynamic data function executes again. In fact, if there are multiple graphs on the same page using the same dynamic data source then the loading function executes _multiple_ times, once for each graph on the page. Hence, if loading your data is a slow operation, your dashboard performance may suffer.
```

Reply:
I removed "This means that your dashboard will always show the very latest data." because I don't think this is true -- it's only the data that was current at the time the dashboard refreshed, and it may update after that (unless the update to the data triggers another refresh?).

Reply:
You are right 👍 it is indeed only the data that's current at the time of refresh - when the data is updated, it doesn't get automatically pushed to the frontend.